crates/sui-graphql-rpc/src/context_data/db_data_provider.rs

```diff
@@ -1377,6 +1397,80 @@ impl PgManager {
             Err(Error::InvalidFilter)
         }
     }
+
+    pub(crate) async fn fetch_dynamic_fields(
+        &self,
+        first: Option<u64>,
+        after: Option<String>,
+        last: Option<u64>,
+        before: Option<String>,
+        address: SuiAddress,
+        filter: Option<DynamicFieldFilter>,
+    ) -> Result<Option<Connection<String, DynamicField>>, Error> {
+        let name_type = filter
+            .and_then(|f| f.name_type)
+            .map(|t| format!("0x2::dynamic_field::Field<%{}%", t));
```

Comment:
> I'll have a separate PR out for dynamic fields without the post-MVP enhancements. 

👍 ❤️ 

> As it is in the db today, the value portion of the dynamic field filter would not need any string matching, but filtering on name would necessitate the use of LIKE.

Yes, I think you're correct, and I like your suggestion of separately storing the `df_name_type` to avoid `LIKE` queries.